airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -915,7 +915,7 @@ def _bq_cast(string_field, bq_type):
     if string_field is None:
         return None
     elif bq_type == 'INTEGER' or bq_type == 'TIMESTAMP':
-        return int(string_field)
+        return int(float(string_field))
```

Comment:
Because Python's `float()` conversion also parses scientific notation, while `int()` not. But I want to keep the type correct in Python so I convert back to int thereafter.

ref: http://stackoverflow.com/questions/29480217/convert-scientific-notation-string-into-int